pkg/apis/storageos/v1/nfsserver_types.go

```diff
@@ -0,0 +1,208 @@
+	//   - Unknown: For some reason the state of the NFS Server could not be
+	//     obtained, typically due to an error in communicating with the host of
+	//     the pod.
+	//
+	Phase string `json:"phase,omitempty"`
+
+	// AccessModes is the access modes supported by the NFS server.
+	AccessModes string `json:"accessModes,omitempty"`
+
+	// TODO(sc): do we want to add more info, e.g. Condition, messages or
+	// StartTime?
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NFSServer is the Schema for the nfsservers API
+// +k8s:openapi-gen=true
+// +kubebuilder:printcolumn:name="status",type="string",JSONPath=".status.phase",description="Status of the NFS server."
+// +kubebuilder:printcolumn:name="capacity",type="string",JSONPath=".spec.resources.requests.storage",description="Capacity of the NFS server."
+// +kubebuilder:printcolumn:name="target",type="string",JSONPath=".status.remoteTarget",description="Remote target address of the NFS server."
+// +kubebuilder:printcolumn:name="access modes",type="string",JSONPath=".status.accessModes",description="Access modes supported by the NFS server."
+// +kubebuilder:printcolumn:name="storageclass",type="string",JSONPath=".spec.storageClassName",description="StorageClass used for creating the NFS volume."
+// +kubebuilder:printcolumn:name="age",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:resource:path=nfsservers,shortName=nfsserver
+type NFSServer struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NFSServerSpec   `json:"spec,omitempty"`
+	Status NFSServerStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NFSServerList contains a list of NFSServer
+type NFSServerList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []NFSServer `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&NFSServer{}, &NFSServerList{})
+}
+
+// ExportsSpec represents the spec of NFS exports
+type ExportsSpec struct {
+	// Name of the export
+	Name string `json:"name,omitempty"`
+
+	// The NFS server configuration
+	Server ServerSpec `json:"server,omitempty"`
+
+	// PVC from which the NFS daemon gets storage for sharing
+	PersistentVolumeClaim corev1.PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
+}
+
+// ServerSpec represents the spec for configuring the NFS server
+type ServerSpec struct {
+	// Reading and Writing permissions on the export
+	// Valid values are "ReadOnly", "ReadWrite" and "none"
+	AccessMode string `json:"accessMode,omitempty"`
+
+	// This prevents the root users connected remotely from having root privileges
+	// Valid values are "none", "rootid", "root", and "all"
+	Squash string `json:"squash,omitempty"`
+
+	// The clients allowed to access the NFS export
+	AllowedClients []AllowedClientsSpec `json:"allowedClients,omitempty"`
```

Comment:
We should either remove `AllowedClients` or figure out how it might be used and test it.  I think it's probably best to remove it for now.

Reply:
Keeping it for now. Will remove in a follow-up PR.

Reply:
Removed it in this PR.